### PR TITLE
Fix edit device bug

### PR DESCRIPTION
**Pull request by kapil-ms**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kapil-ms__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (kaagarw). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs [hunk 1]
```diff
@@ -499,8 +499,12 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
             MethodInfo getMethod;
             int nonediableOrdering;
             DevicePropertyMetadata propertyMetadata;
+            PropertyType propertyType;
 
-            Debug.Assert(deviceProperties != null, "deviceProperties is a null reference.");
+            if (deviceProperties == null)
+            {
+                throw new ArgumentNullException("deviceProperties is a null reference.");
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: throw this instead: DeviceRequiredPropertyNotFoundException ?

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs [hunk 2]
```diff
@@ -710,47 +653,10 @@ private static IEnumerable<DevicePropertyMetadata> GetDevicePropertyConfiguratio
             {
                 IsDisplayedForRegisteredDevices = true,
                 IsDisplayedForUnregisteredDevices = false,
-                IsEditable = false,
+                IsEditable = true,
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: are you sure these are editable? doesn't make sense that location of the device is editable...

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs (hunk 1 above)
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: this exception should be thrown when 

> Exception thrown when required device properties are not found

this is not the case here I think, its just the calling method passed a null argument

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs (hunk 2 above)
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: yes they are , in master

**Pull request merged**